server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -152,6 +156,7 @@ public RepositoriesService(
             threadPool.relativeTimeInMillisSupplier()
         );
         this.preRestoreChecks = preRestoreChecks;
+        this.snapshotMetrics = new SnapshotMetrics(meterRegistry, this::getSnapshotsInProgress);
```

Comment:
We only create `SnapshotMetrics` here so we can provide it with the supplier for the "snapshots in progress" gauge.